src/app/TideIndexApplication.cpp

```diff
@@ -1,1183 +1,1207 @@
+  header_no_mods.mutable_peptides_header()->set_decoys(decoy_type);
+
+  pb::Peptide pbPeptide;
+  unsigned long count = 0;
+  unsigned long numTargets = 0;
+  unsigned long numDuplicateTargets = 0;
+  unsigned long peptide_cnt = 0;
+  
+  if (!sort_on_disk && peptide_list.size() == 0)
+    carp(CARP_FATAL, "No peptides were generated.");
+
+  ifstream sortedFile(pathSortedPeptideFile);
+  unsigned long numLines = 0;
+  TideIndexPeptide* currentPeptide;
+  TideIndexPeptide* duplicatedPeptide;
+  // Filter peptides and keep the unique target peptides and gather the 
+  // location of the peptide in other protein sequences 
+  if (sort_on_disk) {
+    currentPeptide = getNextPeptide(sortedFile, vProteinHeaderSequence);  // get the first peptide  
+  } else {
+    currentPeptide = &(peptide_list[peptide_cnt++]);  // get the first peptide  
+  }
+  if (1==1) {  // This is needed because we need to destroy the peptideWriter and pbAuxLoc later. Ugly solution :/
+    // Create the auxiliary locations header and writer
+    pb::Header auxLocsHeader;
+    auxLocsHeader.set_file_type(pb::Header::AUX_LOCATIONS);
+    pb::Header_Source* auxLocsSource = auxLocsHeader.add_source();
+    auxLocsSource->set_filename(peptidePbFile);
+    auxLocsSource->mutable_header()->CopyFrom(header_no_mods);
+    HeadedRecordWriter auxLocWriter(auxLocsPbFile, auxLocsHeader);  
+    pb::AuxLocation pbAuxLoc;
+    int auxLocIdx = -1;
+    
+    HeadedRecordWriter peptideWriter(peptidePbFile, header_no_mods); // put header in outfile	  
+    while (currentPeptide != nullptr) {
+
+      while (true) {
+
+        if (sort_on_disk) {
+          duplicatedPeptide = getNextPeptide(sortedFile, vProteinHeaderSequence);  // get a peptide  
+          if (duplicatedPeptide == nullptr) {
+            break;
+          }
+        } else {
+          if (peptide_cnt >= peptide_list.size()){
+            duplicatedPeptide = NULL;
+            break;
+          }
+          duplicatedPeptide = &(peptide_list[peptide_cnt++]);  // get a peptide  
+        }
+        if( (*duplicatedPeptide) == (*currentPeptide)) {
+
+          numDuplicateTargets++;
+          carp(CARP_DEBUG, "Skipping duplicate %s.", currentPeptide->getSequence().c_str());
+          pb::Location* location = pbAuxLoc.add_location();
+          location->set_protein_id(duplicatedPeptide->getProteinId());
+          location->set_pos(duplicatedPeptide->getProteinPos());
+          if (sort_on_disk){ 
+            delete duplicatedPeptide;
+          }
+        } else {
+          break;
+        }
+        if (duplicatedPeptide->getMass() < currentPeptide->getMass()){  // Check if sorting worked properly.
+          carp(CARP_INFO, "peptide mass: %lf, subsequent peptide mass %lf", currentPeptide->getMass(), duplicatedPeptide->getMass());
+          carp(CARP_FATAL, "Peptides are not sorted correctly. Sorting seems to be failed. Try again and check the free disk space.");

```

Comment:
"Peptide sorting failed. Please check how much disk space you have available."

Is there an option to specify a different location to use for storing the sorted lists? If so, maybe mention that in this comment.